approach_mission.md

```diff
@@ -0,0 +1,127 @@
+
+Now we have new entity: People.  Let's call it “User”.
+
+Is “user” has relationship with “topic”?
+
+Well, user can register to many topics and many users can register to the same topic, so we will get something like this:
+
+![](6.jpg)
+
+
+Try to answer the “million question” for this case… a bit complicated, huh?! It can be a million users and a million topics...
+So, to make it simple, we can add more entity - let’s call it “Membership”, and it will represent a specific register of user to a topic.
+
+![](7.jpg)
+
+In that case “Membership” refer to “Topic” (because we don’t want “Topic” refer to million “Membership”) and ““Membership” also refer to “User” (because we don’t want “User” refer to million “Membership”). It looks like this:
+
+![](88.jpg)
+
+
+
+Let us continue with our story - 
+
+**Registration expired after a year. The user will get a reminder emails - 3 month, 1 month and 1 day before expired, so he can renew his registration.** 
+
+What information do we need in order to send the reminder emails?
+We need to know of course when the membership began, then we can calculate when 9 months has passed and send the first reminder email. In technical words, the beginning date (or any other date represent something occurs in the system), calls “time stamp.
+
+But this information is not enough. Think about situation user decide not to renew his registration, what happened to his membership? Is it going away, delete from the system? Well, usually we don’t delete content from the system, but we can mark it as inactive.
+So in order to send the user reminder, we need to know two things:  the membership’s created date (i.e "time stamp") and if it is active or not (i.e "status").
+
+![](9.jpg)
+
+Ok, so now we have all the information we need for sending the first reminder email (the one that comes after 9 month), so in order to know to which user we need to send that email, we need to retrieve from the database the right user. 
+Let’s describe in words the query for getting the right information from database:
+
+**Give me all “membership” that their “status” is *active* and their “time stamp” is *today's date minus 9 month***
+
+Ok, we almost there, but there is still one more thing - we need to check that we don't send the email more then one time to the same user. Assuming that we have a lots of "membership" in our database, so we need to limit the number of "membership" we get every time we run the query (if the system will bring us all the fit "membership" at ones, it can be run out of memory). So we will tell the system to brings us only 100 "membership" at a time, and we will run the query every 5 min. 
+How we make sure that the system won't bring us the same "membership" we already sent email for them 5 min ago?
+Of course we need to save an Email log, and check it every time we want to send reminder email.
+
+so let's add new entity call "email log"
```

Comment:
if you have machine names such as email log I think it's better to highlight as code (i.e. `email log`)